go/api/base_client.go

```diff
@@ -512,3 +514,113 @@ func (client *baseClient) RPush(key string, elements []string) (Result[int64], e
 
 	return handleLongResponse(result)
 }
+
+func (client *baseClient) SAdd(key string, members []string) (Result[int64], error) {
+	result, err := client.executeCommand(C.SAdd, append([]string{key}, members...))
+	if err != nil {
+		return CreateNilInt64Result(), err
+	}
+
+	return handleLongResponse(result)
+}
+
+func (client *baseClient) SRem(key string, members []string) (Result[int64], error) {
+	result, err := client.executeCommand(C.SRem, append([]string{key}, members...))
+	if err != nil {
+		return CreateNilInt64Result(), err
+	}
+
+	return handleLongResponse(result)
+}
+
+func (client *baseClient) SMembers(key string) (map[Result[string]]struct{}, error) {
+	result, err := client.executeCommand(C.SMembers, []string{key})
+	if err != nil {
+		return nil, err
+	}
+
+	return handleStringSetResponse(result)
+}
+
+func (client *baseClient) SCard(key string) (Result[int64], error) {
+	result, err := client.executeCommand(C.SCard, []string{key})
+	if err != nil {
+		return CreateNilInt64Result(), err
+	}
+
+	return handleLongResponse(result)
+}
+
+func (client *baseClient) SIsMember(key string, member string) (Result[bool], error) {
+	result, err := client.executeCommand(C.SIsMember, []string{key, member})
+	if err != nil {
+		return CreateNilBoolResult(), err
+	}
+
+	return handleBooleanResponse(result)
+}
+
+func (client *baseClient) SDiff(keys []string) (map[Result[string]]struct{}, error) {
+	result, err := client.executeCommand(C.SDiff, keys)
+	if err != nil {
+		return nil, err
+	}
+
+	return handleStringSetResponse(result)
+}
+
+func (client *baseClient) SDiffStore(destination string, keys []string) (Result[int64], error) {
+	result, err := client.executeCommand(C.SDiffStore, append([]string{destination}, keys...))
+	if err != nil {
+		return CreateNilInt64Result(), err
+	}
+
+	return handleLongResponse(result)
+}
+
+func (client *baseClient) SInter(keys []string) (map[Result[string]]struct{}, error) {
+	result, err := client.executeCommand(C.SInter, keys)
+	if err != nil {
+		return nil, err
+	}
+
+	return handleStringSetResponse(result)
+}
+
+func (client *baseClient) SInterCard(keys []string) (Result[int64], error) {
+	result, err := client.executeCommand(C.SInterCard, append([]string{strconv.Itoa(len(keys))}, keys...))
+	if err != nil {
+		return CreateNilInt64Result(), err
+	}
+
+	return handleLongResponse(result)
+}
+
+func (client *baseClient) SInterCardLimit(keys []string, limit int64) (Result[int64], error) {
+	args := slices.Concat([]string{utils.IntToString(int64(len(keys)))}, keys, []string{"LIMIT", utils.IntToString(limit)})
```

Comment:
You can consider using Go's built-in append function here . This could look like:
``` 
args := append([]string{IntToString(int64(len(keys)))}, keys...)
args = append(args, "LIMIT", IntToString(limit))
``` 
So with that you don't have to rely on  a third-party utility like slices.Concat for argument concatenation.

Reply:
@MikeMwita `slices` is a [standard library](https://pkg.go.dev/std) not a third-party.